plugin/core/registry.py

```diff
@@ -161,3 +168,49 @@ def run_async() -> None:
 class LspRecheckSessionsCommand(sublime_plugin.WindowCommand):
     def run(self, config_name: Optional[str] = None) -> None:
         sublime.set_timeout_async(lambda: windows.lookup(self.window).restart_sessions_async(config_name))
+
+
+def navigate_diagnostics(view: sublime.View, point: Optional[int], forward: bool = True) -> None:
+    try:
+        uri = uri_from_view(view)
+    except MissingUriError:
+        return
+    window = view.window()
+    if not window:
+        return
+    diagnostics = []  # type: List[Diagnostic]
+    for session in windows.lookup(window).get_sessions():
+        diagnostics.extend(session.diagnostics_manager.diagnostics_by_document_uri(uri))
+    if not diagnostics:
+        return
+    # Sort diagnostics by location
+    diagnostics.sort(key=lambda d: operator.itemgetter('line', 'character')(d['range']['start']), reverse=not forward)
+    if point is None:
+        region = first_selection_region(view)
+        point = region.b if region is not None else 0
+    # Find next/previous diagnostic or wrap around and jump to the first/last one, if there are no more diagnostics in
+    # this view after/before the cursor
+    op_func = operator.gt if forward else operator.lt
```

Comment:
This is subjective. (if others do not have problems with following this code, skip this comment)

This code is a little bit harder to follow
do you see a way to simplify this:

```
op_func = operator.gt if forward else operator.lt
for diagnostic in diagnostics:
        diag_pos = point_to_offset(Point.from_lsp(diagnostic['range']['start']), view)
        if op_func(diag_pos, point):
            break
```

Reply:
Ok, It was a little bit hard for me because this is the first time seeing a for/else syntax and the operator module :) 
but I am ok with the code after staring at it for a little bit longer. 

And it makes sense why you did it in this way.

Reply:
It's just a different way of writing

```python
for diagnostic in diagnostics:
    diag_pos = point_to_offset(Point.from_lsp(diagnostic['range']['start']), view)
    if forward:
        if diag_pos > point:
            break
    else:
        if diag_pos < point:
            break
```

but with fewer lines and indentation, and it allows to move one "if" to outside of the loop.

The for/else syntax is actually one of my favorite aspects of Python and I use it whereever I can :)